src/modules/functions.js

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable no-loop-func */
+/* eslint-disable no-restricted-globals */
+
+let listTasks = [];
+// for taking the value from local storage storing in above array
+
+if (localStorage.getItem('listTasks') !== null) {
+  listTasks = JSON.parse(localStorage.getItem('listTasks'));
+}
+// DOM manipulation
+const addbtn = document.querySelector('.add');
+const msg = document.querySelector('.msg');
+const txtField = document.getElementById('taskTextField');
+const taskContainerFromhtml = document.querySelector('.tasks');
+const containerAllFromHtml = document.querySelector('.container');
+let containerEdit;
+let cancelButton;
+let saveButton;
+let inputForEdit;
+let task;
+let removeButton = [];
+// for adding new task to array and localStorageArray
+function add(arr) {
+  if (localStorage.getItem('listTasks') !== null) {
+    arr = JSON.parse(localStorage.getItem('listTasks'));
+  }
+  arr.push({
+    isCompleted: false,
+    index: arr.length,
+    description: txtField.value,
+  });
+  localStorage.setItem('listTasks', JSON.stringify(arr));
+}
+
+// For displaying all the Task in the Screen.
+function display(arr) {
```

Comment:
Nice work on coming this far.

There are still some changes to be made.

- On the initial render of the app, the indexes begin at zero, it should begin at 1. So in your add function, let index be `arr.length + 1`.
- When one is deleted, an index is duplicated, the fix above should also fix this.

<img width="1315" alt="image" src="https://github.com/ZabihullahNooriWardak/brandNewTodoApp/assets/27709832/3cf512d8-3d67-4a9f-815d-2b841bb122d2">